dessia_common/report.py

```diff
@@ -0,0 +1,199 @@
+"""
+report for dessia_common
+
+"""
+from dessia_common.core import DessiaObject
+from typing import List
+from string import Template
+import time
+
+dessia_object_markdown_template = Template(
+'''
+$log
+''')
+
+class Report(DessiaObject):
+    _standalone_in_db = True
+    # _eq_is_data_eq = True
+    # _non_serializable_attributes = []
+    # _non_data_eq_attributes = ['name']
+    # _non_data_hash_attributes = ['name']
+
+    def __init__(self, name_report: str = 'Output',
+                 width_line: int = 100,
+                 # time_start: float = None,
+                 content: str = '',
+                 last_offset: int = 0, #Perhaps we could delete it, to see with platform integration
+                 name: str = ''):
+        """
+        Name and parameters for report's layout
+        """
+
+        #Mettre en attribut facultatif pour l'export txt à chaque écriture, same pour le print
+
+        DessiaObject.__init__(self, name=name)
+        self.width_line = width_line
+        self.name_report = name_report
+        # if time_start is None:
+        self.time_start = time.time()
+        # else:
+        #     self.time_start = time_start
+        self.last_offset = last_offset
+        self.error = False
+        self.content = content
+
+    def add_time(self, offset: int = 0, text_to_add: str = '', time_to_add: float = 0):
+        line = ' ' * offset
+        time_elapsed = str(round((time.time() - self.time_start) / 60, 2))
+        if time_to_add != 0 and text_to_add != '':
+            # line += '*ELAPSED TIME min -- {} {} -- global {}*'.format(text_to_add, time_to_add, time_elapsed)
+            line += f'*ELAPSED TIME min -- {text_to_add} {time_to_add} -- global {time_elapsed}*'
+        else:
+            # line += '*ELAPSED TIME min -- global {}* '.format(time_elapsed)
+            line += f'*ELAPSED TIME min -- global {time_elapsed}* '
+        self.add_lines([line], nb_line_blank=0)
+
+    def open(self, option: str = 'a'):
+        #To avoid : Using open without explicitly specifying an encoding
+        # file = open(self.name_report + '.log', option)
+        # TODO : Mettre with open
+        if option == 'r':
+            file = open(self.name_report + '.log', 'r')
+        elif option == 'w':
+            file = open(self.name_report + '.log', 'w')
+        return file
+
+    def close(self, file):
+        file.close()
+
+    def add_lines(self, lines: List[str], offset: int = 0, nb_line_blank: int = 0):
+        for line in range(nb_line_blank):
+            lines.append('')
+        for line in lines:
+            self.content += offset*'' + line + '  \n  '
```

Comment:
Are you sure that `offset * ''` does something because it is strictly equal to `''` ?

I think these are equivalent:
`self.content += '  \n  '.join(lines)[6:]` 
or
`self.content += lines[0] + '  \n  '.join(lines)`

Reply:
Done by adding a space in the initial ''